Docs 2146 business events v4.2 (#893)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/business-events.adoc

> ifndef::env-site,env-github[]

Reviewer: This topic precedes the previous ones in the actual doc, right?

include::_attributes.adoc[]
endif::[]
:page-aliases: business-events-best-practices.adoc

> Mule applications can collect business event information about flows and message processors that handle your business transactions. This information can include transaction execution time, errors, results (successful completion or failure), and message payload information.

Reviewer: Suggestion for introducing events and transactions, leaving details for event types topic:
```suggestion
Mule applications collect _business event_ information (such as transaction execution time, errors, success or failure results, and message payload information) about flows and message processors that handle your _business transactions_:
* Events map to message processors and connectors and are the low-level details of a transaction. Events are not supported in batch processing.
* Transactions are logical groupings of related events that often correspond to a business view of the system, at the flow level.
```

> Transactions are logical groupings of related events that often correspond to a business view of the system. These are represented by event groupings at the flow level.

Reviewer: See suggested reorder above.

> Events map to message processors and connectors and are the low-level details of a transaction. Events can also be exceptions and custom events. All events that are not custom, are considered default events.

Reviewer: ditto

Reviewer: This introduction to the topic set should also prepare the audience for the subtopics of BE types, monitoring, and other best practices.

[NOTE]

> Batch processors do not support the use of Business Events.

Reviewer: Suggested moving to previous text.

== Business Event Types

> There are two types of business events that you can track: Default events and Custom events.

Reviewer: ```suggestion
You can track two types of business events: _default events_ and _custom events_.
```

> Default event tracking is supported by all connectors and selected message processors. Enabling default events for these elements enables you to perform advanced debugging at runtime. Default event tracking can be enabled at the flow level or the message processor level. +

Reviewer: ```suggestion
Default event tracking is supported by all connectors and selected message processors. Enabling default events for these elements enables you to perform advanced debugging at runtime. You can enable default event tracking at either the flow level or the message processor level. +
```

See xref:business-events-in-components.adoc[Default Event Tracking] for configuration details.

> Custom event tracking is supported by the Custom Business Event Component. Custom events are always tracked once configured. Custom events enable you to track high-level activities in your flow that are relevant to your business. +

Reviewer: ```suggestion
Custom event tracking is supported by the Custom Business Event component. Custom events are always tracked, enabling you to monitor high-level activities in your flow that are relevant to your business. +
```

See xref:business-events-custom.adoc[Custom Business Event Component] for configuration details.

== Monitoring Business Events

> You can enable Insight in Runtime Manager to monitor business events at runtime. These events can help you analyze the root cause of failures, isolate performance bottlenecks, and test for compliance with company procedures.

Reviewer: I was not able to find "Insight" in the official product name list, although I did see CloudHub Insight in some old doc. Is this name still relevant?

> Certain default events are tracked automatically once Insight is enabled, but others need to be enabled in your application before they are tracked.

Reviewer: ```suggestion
Certain default events are tracked automatically after you enable Insight, but you must enable others in your application before they are tracked.
```

> == Best Practices For Business Events

Reviewer: ```suggestion
== Best Practices for Business Events
```

Reviewer: ...but I think a more specific heading is probably "Requirements for Tracking Business Events."

Reviewer: ...followed by a transition sentence that ends with a colon, to lead in to the list.

> * Enable default events only for processes that have particular value to you. Determine which stages within a business transaction you need to track, and enable tracking for those stages before deployment. +

Reviewer: ```suggestion
* Enable default events only for processes that have particular value to you. 
To filter out unnecessary information, determine which stages within a business transaction you need to track, and enable tracking for only those stages.+
```

> Tracking all possible events is also an option, but this generates unnecessary info that you have to filter or query to find the important events you need to analyze.

Reviewer: OMIT per previous suggestion.

> * Use custom events to track key process indicators. For example: "Total Order Amount" or "Tracking Number" to surface the high-level business activities in your flow.

Reviewer: ```suggestion
* Use custom events to track key process indicators: for example, "Total Order Amount" or "Tracking Number" to surface the high-level business activities in your flow.
```

> * Customize the transaction ID so that meaningful information, such as an order number, an employee identification number, or a shipment tracking number, is displayed for a transaction. This practice makes analysis and debugging easier and more intuitive at runtime, whether you are using Mule Management Console or CloudHub.

Reviewer: ```suggestion
* Customize the transaction ID so that meaningful information, such as an order number, an employee identification number, or a shipment tracking number, is displayed for a transaction.
This practice makes analysis and debugging easier and more intuitive at runtime, whether you are using Mule Management Console or CloudHub.
```
